### PR TITLE
Update our pre-commit hook versions

### PR DESCRIPTION
**Pull request by stefan-anmut**
This PR addresses the tech debt built up on our pre-commit hooks in the last couple of years. Pre-commit hook versions have been updated to their latest versions. This PR has been tried and tested ✅

### CODE REVIEW

File: {{cookiecutter.repo_name}}/.pre-commit-config.yaml [hunk 1]
```diff
@@ -77,7 +77,7 @@ repos:
       done"
 
 - repo: https://github.com/anmut-consulting/pipenv-setup
-  rev: v3.1.5
+  rev: v3.1.6
```
Comment (stefan-anmut) [inline {{cookiecutter.repo_name}}/.pre-commit-config.yaml:80]: WARNING: I've updated the version of our pipenv-setup hook in anticipation of merging into master [this PR](https://github.com/anmut-consulting/pipenv-setup/pull/7).

File: {{cookiecutter.repo_name}}/.pre-commit-config.yaml [hunk 2]
```diff
@@ -17,7 +17,7 @@ repos:
     exclude: docs/|setup.cfg
 
 - repo: https://github.com/myint/docformatter
```
Comment (csut-anmut) [inline {{cookiecutter.repo_name}}/.pre-commit-config.yaml:19]: The given url (https://github.com/myint/docformatter) appears to redirect to https://github.com/PyCQA/docformatter which is the correct one, I believe. While the redirection appears to be working, perhaps for clarity this url should be updated?

File: {{cookiecutter.repo_name}}/.pre-commit-config.yaml [hunk 3]
```diff
@@ -48,18 +48,18 @@ repos:
         args: ['--ignore=E203,E266,E501,W503', '--max-line-length=88', '--max-complexity=18', '--select=B,C,E,F,W,T4']
 
 - repo: https://github.com/timothycrosley/isort
```
Comment (csut-anmut) [inline {{cookiecutter.repo_name}}/.pre-commit-config.yaml:50]: Same as above, the offical repo appears to have been brought under the PyCQA umbrella https://github.com/PyCQA/isort, so perhaps we update the URL to match?

**Review by csut-anmut — COMMENTED** [review]
Some potential URL updates to be made. Looks good otherwise. Thanks for your work on this @stefan-anmut

File: {{cookiecutter.repo_name}}/.pre-commit-config.yaml (hunk 2 above)
Comment (stefan-anmut) [reply {{cookiecutter.repo_name}}/.pre-commit-config.yaml]: Thanks Chris, great spot 👍

**Review by csut-anmut — APPROVED** [review]
Good work 👍

**Review by jmartin-anmut — APPROVED** [review]
Looks good. Nice URL spots from Chris too. Good work both of you

**Pull request merged**